SEP-0010.md

```diff
@@ -0,0 +1,67 @@
+# SEP-0010 -- Handling broken or missing metadata
+
+| SEP           | 0010                             |
+|---------------|----------------------------------|
+| title         | Handling broken or missing metadata  |
+| author(s)     | [David Stansby](https://orcid.org/0000-0002-1365-1908)<br>David Stansby           |
+| contact email | dstansby@gmail.com               |
+| date-creation | 22-09-2020                       |
+| type          | standard                         |
+| discussion    | https://github.com/sunpy/sunpy/issues/4410  |
+| status        | discussion                       |
+
+# Introduction
+Although standards exist (e.g. the FITS standard) to describe metadata in data
+files, it is not un-common for data sources to not comply with these standards.
+This SEP lays out how sunpy core should behave when it loads data with
+non-standards-compliant metadata.
+
+# Detailed Description
+
+## Proposal
+This SEP proposes the following process for handling metadata:
+
+1. Generic data loaders
+
+The sunpy core package contains base types for storing data, e.g. GenericMap,
+GenericTimeseries. Trying to load data that is not standards compliant with
+these types should raise an error as soon as possible in the data loading
+process. This allows users and developers to be easily notified of non-compliant
+metadata, and follows fail fast approach.
```

Comment:
I wonder if the default mode should be error, but if an there could/should be a keyword that either changes exception to warning or allows non-compliant metadata to be loaded.

I can easily conceive of situations where there is older data to be loaded that may not have come through a standard instrument pipeline, or that there is not an instrument specific class for.  Under this framework, if that data has noncompliant metadata, there is no mechanism to load it or to force it to be loaded.

An alternative would be to provide a `NoncompliantMap` interface.  Either way, the user would be forced to explicitly acknowledge that their data is not compliant and SunPy would not be obligated to do anything "correct" with it.

Reply:
The current mechanism for manually fixing metadata is documented here: https://docs.sunpy.org/en/stable/code_ref/map.html#fixing-map-metadata Essentially, load the data with a fits reader, fix the metadata, and then give it to `Map`. I think this is a reasonable tradeoff, and if fixes are common enough a new map class can be created.